### ISSUE TITLE
Per-input xkb configuration

### ISSUE THREAD

**Issue by deifactor** [ifreund/river#400]
I type in Colemak and use both an external keyboard and the internal keyboard on my laptop. The internal keyboard needs xkb to switch to Colemak, but the external one just natively generates the right keycode. It'd be nice to swap xkb state based on the input device like sway or hikari does so that I don't try typing in Colemak with the Colemak layout enabled.

**Comment by deifactor** [comment]
Actually, wait, I hadn't tested this before, but it... works as-is right now? Hitting my layout toggle bind doesn't mess with my internal keyboard, only my external. I have no clue how this is even possible.

**Comment by hillyu** [comment]
same situation here, would be nice to have sway-like input configs for each keyboard based on names/identifiers. 
Good job to the dev btw. Switched from sway for the dwm style tiling, have to say this is wm is more polished than described.

**Comment by hillyu** [comment]
hey, want to update if anyone is interested, I found a solution for this, probabally more general and elegant, that is to use a system wide key remap daemon, like this one: https://github.com/rvaiya/keyd. It supports per keybard settings and can do more than just key remapping. Also it is not tied with display server can even work in console. So just in case anyone has the same need as I do.

**Comment by FoundOne** [comment]
Does keyd work in river? I see sway mentioned which is wlroots based, but  I have no idea how river handles the keyboard layouts. I'm pretty sure that the sway keyboard layout indicator in yambar does not work in river. Can someone confirm whether keyd works in river? That what bothers me is the lack of a proper way to know which layout are you using, so I have to check it every time.

**Comment by ifreund** [comment]
@FoundOne Yes, keyd should work with river and all other wayland compositors. It is implemented a layer below river in the input stack.
